specifyweb/frontend/js_src/lib/utils/ajax/throttledPromise.ts

```diff
@@ -10,8 +10,8 @@ const currentRequestsGenerator = <T>(): WritableArray<PromiseWithSpec<T>> => [];
 
 let maybeFulfilled: WritableArray<KillablePromise<any>> = [];
 
-export const cleanThrottledPromises = (): void => {
-  maybeFulfilled.forEach((promise) => promise.killPromise());
+export const cleanThrottledPromises = (killPromise = true): void => {
```

Comment:
Maybe `killPromises` might be a better name for the parameter name? 
We are not killing one promise, we are potentially killing multiple.

Reply:
You do kill a single promise when you do promise.killPromise(). It's just that you do it for every single one of them.

Reply:
Sure, but some to a consumer of the function that distinction is not made clear until they look at the implementation. 
I think `killPromises` makes the intention behind the parameter immediately clear: you put in a truthy value, then multiple promises will be killed.

Ultimately, this is just semantics. The function implementation should only be a click (or hover) away from most developers :shrug: